tests/build-rss.test.js

```diff
@@ -0,0 +1,182 @@
+    } catch (err) {
+      error = err
+    }
+
+    const filePath = path.join(__dirname, '..', 'public', outputPath);
+    const fileContent = fs.readFileSync(filePath, 'utf8');
+
+    const itemTitles = fileContent.match(/<title>(.*?)<\/title>/g);
+
+    expect(error).toBeUndefined();
+    expect(itemTitles[1]).toContain('Test Post 1');
+    expect(itemTitles[2]).toContain('Another Featured Post');
+    expect(itemTitles[3]).toContain('Non-Featured Post 1');
+  });
+
+  it('should sort posts by date in descending order', async () => {
+    jest.doMock('../config/posts.json', () => mockRssData, { virtual: true });
+
+    let error;
+    try {
+      await rssFeed(type, title, desc, outputPath);
+    } catch (err) {
+      error = err
+    }
+
+    const filePath = path.join(__dirname, '..', 'public', outputPath);
+    const fileContent = fs.readFileSync(filePath, 'utf8');
+
+    const itemTitles = fileContent.match(/<title>(.*?)<\/title>/g);
+
+    expect(error).toBeUndefined();
+    expect(itemTitles[1]).toContain('Test Post 1');
+    expect(itemTitles[2]).toContain('Another Featured Post');
+    expect(itemTitles[3]).toContain('Non-Featured Post 1');
+    expect(itemTitles[4]).toContain('Non-Featured Post 3');
+    expect(itemTitles[5]).toContain('Non-Featured Post 2');
+  });
+
+  it('should set correct enclosure type based on image extension', async () => {
+    jest.doMock('../config/posts.json', () => mockRssData, { virtual: true });
+
+    let error;
+    try {
+      await rssFeed(type, title, desc, outputPath);
+    } catch (err) {
+      error = err
+    }
+
+    const filePath = path.join(__dirname, '..', 'public', outputPath);
+    const fileContent = fs.readFileSync(filePath, 'utf8');
+
+    expect(error).toBeUndefined();
+    expect(fileContent).toContain('<enclosure url="https://www.asyncapi.com/img/test-cover.png"');
+    expect(fileContent).toContain('type="image/png"');
+    expect(fileContent).toContain('<enclosure url="https://www.asyncapi.com/img/test-cover.svg"');
+    expect(fileContent).toContain('type="image/svg+xml"');
+    expect(fileContent).toContain('<enclosure url="https://www.asyncapi.com/img/test-cover.webp"');
+    expect(fileContent).toContain('type="image/webp"');
+  });
+
+  it('should catch and handle errors when write operation fails', async () => {
+    jest.doMock('../config/posts.json', () => mockRssData, { virtual: true });
+
+    const invalidOutputPath = "invalid/path";
+
+    let error;
+    try {
+      await rssFeed(type, title, desc, invalidOutputPath);
+    } catch (err) {
+      error = err;
+    }
+
+    expect(error.message).toMatch(/ENOENT|EACCES/);
+  });
+
+  it('should throw an error when posts.json is not found', async () => {
+    const postsJsonPath = path.join(__dirname, '..', 'config', 'posts.json');
+    const tempPath = path.join(__dirname, '..', 'config', 'posts.json.temp');
+    
+    if (fs.existsSync(postsJsonPath)) {
+      fs.renameSync(postsJsonPath, tempPath);
+    }
```

Comment:
Ideally, you shouldn't modify the actual files. In fact, you can remove this whole test since `build-rss` functional tests has nothing to do with the `getAllPosts` function.